flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/checkpoints/CheckpointTriggerInfo.java

```diff
@@ -0,0 +1,78 @@
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.rest.messages.checkpoints;
+
+import org.apache.flink.runtime.rest.messages.ResponseBody;
+import org.apache.flink.runtime.rest.messages.json.SerializedThrowableDeserializer;
+import org.apache.flink.runtime.rest.messages.json.SerializedThrowableSerializer;
+import org.apache.flink.util.SerializedThrowable;
+
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonCreator;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonInclude;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.annotation.JsonSerialize;
+
+import javax.annotation.Nullable;
+
+import static org.apache.flink.util.Preconditions.checkArgument;
+
+/** Represents information about a finished triggered checkpoint. */
+@JsonInclude(JsonInclude.Include.NON_NULL)
+public class CheckpointTriggerInfo implements ResponseBody {
+
+    private static final String FIELD_NAME_CHECKPOINT_ID = "checkpointId";
+
+    private static final String FIELD_NAME_FAILURE_CAUSE = "failure-cause";
+
+    @JsonProperty(FIELD_NAME_CHECKPOINT_ID)
+    @Nullable
+    private final Long checkpointId;
+
+    @JsonProperty(FIELD_NAME_FAILURE_CAUSE)
+    @JsonSerialize(using = SerializedThrowableSerializer.class)
+    @JsonDeserialize(using = SerializedThrowableDeserializer.class)
+    @Nullable
+    private final SerializedThrowable failureCause;
+
+    @JsonCreator
+    public CheckpointTriggerInfo(
+            @JsonProperty(FIELD_NAME_CHECKPOINT_ID) @Nullable final Long checkpointId,
+            @JsonProperty(FIELD_NAME_FAILURE_CAUSE)
+                    @JsonDeserialize(using = SerializedThrowableDeserializer.class)
+                    @Nullable
+                    final SerializedThrowable failureCause) {
+        checkArgument(
+                checkpointId != null ^ failureCause != null,
+                "Either checkpointId or failureCause must be set");
+
+        this.checkpointId = checkpointId;
+        this.failureCause = failureCause;
+    }
+
+    @Nullable
+    public Long getCheckpointId() {
+        return checkpointId;
+    }
+
+    @Nullable
+    public SerializedThrowable getFailureCause() {
+        return failureCause;
+    }
```

Comment:
add `@JsonIgnore`

Reply:
Sure. Do we want to fix [SavepointInfo](https://github.com/apache/flink/blob/99c2a415e9eeefafacf70762b6f54070f7911ceb/flink-runtime/src/main/java/org/apache/flink/runtime/rest/messages/job/savepoints/SavepointInfo.java#L74) as well?